App/Containers/OnboardingScreen/index.tsx

```diff
@@ -72,19 +70,20 @@ class OnboardingScreen extends React.Component<Props, State> {
   }
 
   nextPage = () => {
-    if (this.pages && this.pages.props.children.length - 1 > this.state.currentPage && !this.state.blockNext) {
-      this.setState({blockNext: true})
+    if (this.pages && this.pages.props.children.length - 1 > this.state.currentPage) {
+      this.setState({
+        showArrow: this.showArrowForIndex(this.state.currentPage + 1)
+      })
       this.pages.scrollToPage(this.state.currentPage + 1)
-      setTimeout(() => {
-        this.setState({
-          blockNext: false,
-          showArrow: this.showArrowForIndex(this.state.currentPage + 1),
-          currentPage: this.state.currentPage + 1
-        })
-      }, 350)
     }
   }
 
+  onScrollEnd = () => {
+    this.setState({
+      currentPage: this.state.currentPage + 1
+    })
+  }
+
```

Comment:
Using the `onScrollEnd` callback from the pager component to update `state.currentPage` once we've actually arrived on that page.